File: app/src/main/java/com/song/deviceinfo/info/EmulatorInfo.java
package com.song.deviceinfo.info;

import android.content.Context;
import android.os.Build;

import com.song.deviceinfo.utils.CommandUtils;
import com.song.deviceinfo.utils.EmulatorUtils;

import java.util.ArrayList;
import java.util.List;

import androidx.core.util.Pair;

/**
 * Created by chensongsong on 2020/7/9.
 */
public class EmulatorInfo {

    /**
     * 获取应用列表
     *
     * @param context
     * @return
     */
    public static List<Pair<String, String>> getEmulatorInfo(Context context) {
        List<Pair<String, String>> list = new ArrayList<>();
        list.add(new Pair<>("Su Version", EmulatorUtils.getSuVersion()));
        list.add(new Pair<>("Sound", EmulatorUtils.getSound()));
        list.add(new Pair<>("Battery", EmulatorUtils.getBatteryInfo(context)));
        list.add(new Pair<>("BrandInfo", EmulatorUtils.getBrandInfo()));
        list.add(new Pair<>("Launcher", EmulatorUtils.getLauncherPackageName(context)));
        list.add(new Pair<>("ThermalCount", EmulatorUtils.thermalCheck() + ""));
        list.add(new Pair<>("CameraCheck", EmulatorUtils.cameraCheck(context) + ""));
        list.add(new Pair<>("RSSI", NetWorkInfo.getRssi(context)));
        list.add(new Pair<>("Abi", CommandUtils.getProperty("ro.product.cpu.abi")));
        list.add(new Pair<>("AbiList", CommandUtils.getProperty("ro.product.cpu.abilist")));
        list.add(new Pair<>("Hardware", CommandUtils.getProperty("ro.hardware")));
        list.add(new Pair<>("Arch", CommandUtils.execute("uname -m")));
        list.add(new Pair<>("BluetoothFile", EmulatorUtils.bluetoothCheck() + ""));
        list.add(new Pair<>("SpecialFiles", EmulatorUtils.specialFilesEmulatorCheck() + ""));
        list.add(new Pair<>("x86", EmulatorUtils.getArch() + ""));
        list.add(new Pair<>("MapsArch", EmulatorUtils.getMapsArch()));
        list.add(new Pair<>("Qemu", EmulatorUtils.qemuCheck() + ""));
        list.add(new Pair<>("Pipes", EmulatorUtils.checkPipes() + ""));
        list.add(new Pair<>("Model Name", EmulatorUtils.getModelName()));
        list.add(new Pair<>("Product", CommandUtils.getProperty("ro.product.name")));
        list.add(new Pair<>("Manufacturer", CommandUtils.getProperty("ro.product.manufacturer")));
        list.add(new Pair<>("Brand", CommandUtils.getProperty("ro.product.brand")));
        list.add(new Pair<>("Device", CommandUtils.getProperty("ro.product.device")));
        list.add(new Pair<>("Model", CommandUtils.getProperty("ro.product.model")));
        list.add(new Pair<>("Fingerprint", Build.FINGERPRINT));
        return list;
    }

}

File: app/src/main/java/com/song/deviceinfo/info/EmulatorInfo.kt
package com.song.deviceinfo.info

import android.content.Context
import android.os.Build
import androidx.core.util.Pair
import com.google.firebase.perf.metrics.AddTrace
import com.song.deviceinfo.utils.CommandUtils.execute
import com.song.deviceinfo.utils.CommandUtils.getProperty
import com.song.deviceinfo.utils.EmulatorUtils.arch
import com.song.deviceinfo.utils.EmulatorUtils.bluetoothCheck
import com.song.deviceinfo.utils.EmulatorUtils.brandInfo
import com.song.deviceinfo.utils.EmulatorUtils.cameraCheck
import com.song.deviceinfo.utils.EmulatorUtils.checkPipes
import com.song.deviceinfo.utils.EmulatorUtils.getBatteryInfo
import com.song.deviceinfo.utils.EmulatorUtils.getLauncherPackageName
import com.song.deviceinfo.utils.EmulatorUtils.getMapsInfo
import com.song.deviceinfo.utils.EmulatorUtils.mapsArch
import com.song.deviceinfo.utils.EmulatorUtils.modelName
import com.song.deviceinfo.utils.EmulatorUtils.qemuCheck
import com.song.deviceinfo.utils.EmulatorUtils.setLauncherInfo
import com.song.deviceinfo.utils.EmulatorUtils.sound
import com.song.deviceinfo.utils.EmulatorUtils.specialFilesEmulatorCheck
import com.song.deviceinfo.utils.EmulatorUtils.suVersion
import com.song.deviceinfo.utils.EmulatorUtils.thermalCheck

/**
 * Created by chensongsong on 2020/7/9.
 */
object EmulatorInfo {
    /**
     * 获取应用列表
     *
     * @param context
     * @return
     */
    @AddTrace(name = "EmulatorInfo.getEmulatorInfo")
    fun getEmulatorInfo(context: Context): List<Pair<String, String>> {
        val list: MutableList<Pair<String, String>> = ArrayList()
        list.add(Pair("Su Version", suVersion))
        list.add(Pair("Sound", sound))
        list.add(Pair("Battery", getBatteryInfo(context)))
        list.add(Pair("BrandInfo", brandInfo))
        val launcher = getLauncherPackageName(context)
        list.add(Pair("Launcher", launcher))
        setLauncherInfo(context, list, launcher)
        list.add(Pair("ThermalCount", thermalCheck().toString() + ""))
        list.add(Pair("CameraCheck", cameraCheck(context).toString() + ""))
        list.add(Pair("RSSI", NetWorkInfo.getRssi(context)))
        list.add(Pair("Abi", getProperty("ro.product.cpu.abi")))
        list.add(Pair("AbiList", getProperty("ro.product.cpu.abilist")))
        list.add(Pair("Hardware", getProperty("ro.hardware")))
        list.add(Pair("Arch", execute("uname -m")))
        list.add(Pair("BluetoothFile", bluetoothCheck().toString() + ""))
        list.add(Pair("SpecialFiles", specialFilesEmulatorCheck().toString() + ""))
        list.add(Pair("x86", arch.toString() + ""))
        list.add(Pair("MapsArch", mapsArch))
        // 新增 ARM 模拟器检测特性
        getMapsInfo(list)
        list.add(Pair("Qemu", qemuCheck().toString() + ""))
        list.add(Pair("Pipes", checkPipes().toString() + ""))
        list.add(Pair("Model Name", modelName))
        list.add(Pair("Product", getProperty("ro.product.name")))
        list.add(Pair("Manufacturer", getProperty("ro.product.manufacturer")))
        list.add(Pair("Brand", getProperty("ro.product.brand")))
        list.add(Pair("Device", getProperty("ro.product.device")))
        list.add(Pair("Model", getProperty("ro.product.model")))
        list.add(Pair("Fingerprint", Build.FINGERPRINT))
        return list
    }
}
